### PR TITLE
Add helper methods to create custom villager types.

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

**Review by modmuss50 — APPROVED** [review]
Looks good, test mod might be useful, not too major here though.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+public final class VillagerTypeHelper {
+	private static final Logger LOGGER = LogManager.getLogger();
+
+	/**
+	 * Creates a new villager type.
+	 *
+	 * @param id the id of the villager type
+	 * @return a new villager type
+	 */
+	public static VillagerType create(Identifier id) {
```
Comment (Player3324) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: IMO plain create implies creating an instance of the surrounding type (VillagerTypeHelper instance)

Elsewhere we used register, but idk if this case has any side effects / does something similar to registration.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Utilities related to the creation of {@link VillagerType}s.
+ * Not to be confused with a {@link VillagerProfession}, a villager type defines the appearance of a villager.
+ *
+ * <p>Creation of custom villager types may be done via the use of a simple helper method.
+ * Below is an example of how one wouldd create a villager type:
+ * <blockquote><pre>
+ * // Note: You should register your villager types via the corresponding registry
+ * public static final VillagerType MOUNTAIN = VillagerTypeHelper.create(new Identifier("mymod", "mountain"));
+ * </pre></blockquote>
```
Comment (Player3324) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: The examples should use the standard combination of `@code` and `<pre>`

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	/**
+	 * Maps a biome to villager type.
+	 * This mapping is used when a new villager entity is created.
+	 *
+	 * @param biomeKey the registry key of the biome
+	 * @param villagerType the villager type
+	 */
+	public static void mapBiomeToVillagerType(RegistryKey<Biome> biomeKey, VillagerType villagerType) {
```
Comment (Player3324) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: `map` is a weak term, it doesn't really explain what happens - could as well mean "give me the villager type for the biome" (ignoring the signature).

The impl suggests that it adds the villager type to a biome ("add" vs "map", "to biome" vs "to villager"). I'd thus call it addVillagerTypeToBiome.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java (hunk 1 above)
Comment (i509VCB) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: I decided to rename this to register and also make it register the villager type.

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import net.minecraft.world.biome.Biome;
+
+import net.fabricmc.fabric.mixin.object.builder.VillagerTypeAccessor;
+
+/**
+ * Utilities related to the creation of {@link VillagerType}s.
+ * Not to be confused with a {@link VillagerProfession}, a villager type defines the appearance of a villager.
+ *
+ * <p>Creation and registration of custom villager types may be done via the use of a simple helper method.
+ * Below is an example of how one wouldd create a villager type:
```
Comment (frqnny) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: Spelling error. "wouldd" -> "would"

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+ * <blockquote><pre>
+ * public static final VillagerType MOUNTAIN = VillagerTypeHelper.register(new Identifier("mymod", "mountain"));
+ * </pre></blockquote>
+ *
+ * <p>Creation and registration of a villager type does not guarantee villagers of a specific type will be created in a world.
+ * Typically the villager type is bound to a specific group of biomes.
+ * Below is an example of allowing creation of mountain villagers in a mountain biome:
```
Comment (frqnny) [inline fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: Maybe, I'd add a short sentence about how textures work, or maybe we can just leave it to yarn to do some javadocs on VillagerType class

**Review by frqnny — COMMENTED** [review]
just a spelling error and something about textures

File: fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java (hunk 5 above)
Comment (i509VCB) [reply fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/api/object/builder/v1/villager/VillagerTypeHelper.java]: Ah yeah I should cover that, I did mention textures in villager profession pull request.

**Review by Player3324 — APPROVED** [review]
Looks good, thanks!

**Review by frqnny — APPROVED** [review]
Great!

**Review by modmuss50 — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by Mysterious-Dev — APPROVED** [review]

**Pull request merged**